scripts/diff_jsons.py

```diff
@@ -15,12 +16,22 @@
     parser.add_argument("first_json", help="first JSON to compare")
     parser.add_argument("second_json", help="second JSON to compare")
     parser.add_argument("--significant-digits", type=int, default=5, help="number of significant digits to use when comparing numeric values")
-    parser.add_argument("--exclude-paths", nargs="+", action=ExtendOverwriteDefault, help="list of paths to exclude from consideration when performing a diff", default=["root['generated_by']['version']"])
+    parser.add_argument("--exclude-paths", nargs="+", action=ExtendOverwriteDefault, help="list of paths to exclude from consideration when performing a diff", default=["root['generated_by']", "root['meta']['updated']"])
     parser.add_argument("--exclude-regex-paths", nargs="+", action="extend", help="list of path regular expressions to exclude from consideration when performing a diff")
     parser.add_argument("--ignore-numeric-type-changes", action="store_true", help="ignore numeric type changes in the diff (e.g., int of 1 to float of 1.0)")
 
     args = parser.parse_args()
 
+    # Test for most fatal errors in regex path usage
+    # Exclude regexes should never match `'`, otherwise the diff is always going to pass
```

Comment:
Should this include `"` as well? I'm assuming `root['generated_by']` and `root["generated_by"]` are both valid.

Reply:
I _think_ that the path that deep diff uses to match against uses a single quote not double quote. Hence excluding only single quote. Does that make sense?

Reply:
I just tested with both `['seqid']` and `["seqid"]`:

```diff
   $ python3 "$TESTDIR/../../../../scripts/diff_jsons.py" "$TESTDIR/../data/clades.json" clades_no-labels.json --exclude-regex-paths "['seqid']"
-  {'dictionary_item_removed': [root['branches']]}
+  Traceback (most recent call last):
+    File "/Users/vlin/Dropbox/repos/nextstrain/augur/tests/functional/clades/cram/../../../../scripts/diff_jsons.py", line 30, in <module>
+      raise Exception(
+  Exception: Exclude regex ['seqid'] matches `'` which means this diff will always pass which is probably not what you want.
+  You probably forgot to escape something in your regex. See for example: https://stackoverflow.com/a/79173188/7483211
+  [1]
```

```diff
   $ python3 "$TESTDIR/../../../../scripts/diff_jsons.py" "$TESTDIR/../data/clades.json" clades_no-labels.json --exclude-regex-paths '["seqid"]'
-  {'dictionary_item_removed': [root['branches']]}
+  {}
```

It looks like `"` should get the same treatment.

Reply:
Ah yeah, there's a bunch more things that should not not be allowed to be matched, things like `[` and `]` and any single letter. But in a way it's not super clearcut as someone might want to exclude everything but a single path (they should use an include path instead then) - so I don't know how far we want to go here. Including `"` sounds good though, no harm I can see.